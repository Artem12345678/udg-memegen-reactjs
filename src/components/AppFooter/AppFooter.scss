.footer {
  padding: 30px 0;

  &__nav {
    display: flex;
    justify-content: space-between;
  }
}

.nav-link {
  &--footer {
    color: #fff;

    &:hover,
    &:focus,
    &:active {
      color: darken(#fff, 10%);
    }
  }
}
