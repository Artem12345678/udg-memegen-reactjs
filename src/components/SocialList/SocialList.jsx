import React, { Component } from "react";
import SocialListItem from "../SocialListItem/SocialListItem";

import "./SocialList.scss";

class SocialList extends Component {
  state = {
    links: [
      {
        url: "https://www.github.com/Artem12345678",
        icon: "github"
      },
      {
        url: "https://www.instagram.com/ystimenkoartem",
        icon: "instagram"
      }
    ]
  };

  render() {
    const { links } = this.state;

    return (
      <ul className="social">
        {links.map(item => (
          <li key={item.icon} className="social__item">
            <SocialListItem url={item.url} icon={item.icon} />
          </li>
        ))}
      </ul>
    );
  }
}

export default SocialList;
