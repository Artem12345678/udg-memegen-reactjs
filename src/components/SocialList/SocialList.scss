.social {
  display: flex;
  list-style-type: none;

  &__item {
    position: relative;
    z-index: 1;
    display: inline-block;
    margin-right: 10px;

    &:last-child {
      margin-right: 0;
    }
  }
}
