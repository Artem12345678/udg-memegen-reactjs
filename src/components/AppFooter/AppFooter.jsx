import React from "react";
import Nav from "react-bootstrap/Nav";
import { Link } from "react-router-dom";
import SocialList from "../SocialList/SocialList";

import "./AppFooter.scss";

const AppFooter = () => {
  return (
    <footer className="footer bg-secondary">
      <div className="container">
        <div className="footer__nav">
          <Nav>
            <Nav.Link as={Link} to="/" className="nav-link--footer">
              Home
            </Nav.Link>
          </Nav>

          <div className="footer__social">
            <SocialList />
          </div>
        </div>
      </div>
    </footer>
  );
};

export default AppFooter;
