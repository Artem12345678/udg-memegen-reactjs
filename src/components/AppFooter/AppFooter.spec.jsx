import React from "react";
import { shallow } from "enzyme";

import AppFooter from "./AppFooter";
import SocialList from "../SocialList/SocialList";

describe("AppFooter.jsx", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AppFooter />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders <SocialList/> component", () => {
    expect(wrapper.find(SocialList).exists()).toBe(true);
  });
});
