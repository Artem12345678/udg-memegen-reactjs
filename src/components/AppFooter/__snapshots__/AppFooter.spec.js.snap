// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.jsx matches snapshot 1`] = `
<footer
  className="footer bg-secondary"
>
  <div
    className="container"
  >
    <div
      className="footer__nav"
    >
      <Nav
        fill={false}
        justify={false}
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          className="nav-link--footer"
          disabled={false}
          to="/"
        >
          Home
        </NavLink>
      </Nav>
      <div
        className="footer__social"
      >
        <SocialList />
      </div>
    </div>
  </div>
</footer>
`;
